subsys/bluetooth/host/conn.c

```diff
@@ -2232,7 +2232,10 @@ static void deferred_work(struct k_work *work)
 		 * auto connect flag if it was set, instead just cancel
 		 * connection directly
 		 */
-		bt_le_create_conn_cancel();
+		if (bt_le_create_conn_cancel() == -ENOBUFS) {
+			LOG_WRN("No buffers to cancel connection, retrying in 10 ms");
+			k_work_reschedule(dwork, K_MSEC(10));
```

Comment:
If something is blocking HCI and thus causing a buffer to be unavailable for e.g. a second, then we'd see a lot of these. Would it make sense to use `LOG_WRN_RATELIMIT`? Or is it better to log them all? 

Should we perform a LOG_ERR or assert if the return value is not `ENOBUFS`?

Reply:
> If something is blocking HCI and thus causing a buffer to be unavailable for e.g. a second, then we'd see a lot of these. Would it make sense to use LOG_WRN_RATELIMIT? Or is it better to log them all?

I think the fact this behavior has been ignored for so long sort of shows that its not a common problem, I wouldn't expect to see a lot of these messages. What situation do you forsee that would result in no buffers being available for a second?

> Should we perform a LOG_ERR or assert if the return value is not ENOBUFS?

Happy to take suggestions. `-ENOBUFS` is an explicit error code returned from `bt_le_create_conn_cancel`, any other errors would come from `bt_hci_cmd_send_sync`.

Reply:
> What situation do you forsee that would result in no buffers being available for a second?

If buffers are unavailable for a second, and the retry timer for this is 10ms, wouldn't we see 100 of these log messages in such a case? 

> -ENOBUFS is an explicit error code returned from bt_le_create_conn_cancel, any other errors would come from bt_hci_cmd_send_sync.

If the controller fails to handle the request, then it would still fail silently with your addition here. I think it would make sense to log such as case as well (with LOG_ERR perhaps)

Reply:
> If buffers are unavailable for a second, and the retry timer for this is 10ms, wouldn't we see 100 of these log messages in such a case?

Sure, my question is why would buffers be unavailable for a second, and why would 100 of these log messages be a large problem if they were? Those other logging functions are not zero overhead, and as previously mentioned I can't imagine this being a common problem since it has not yet been fixed.